### PR TITLE
auth: Remove disable-tcp option

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
Also turns the TCPNameserver into a shared_ptr instead of a naked pointer.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/common_startup.hh [hunk 1]
```diff
@@ -46,7 +46,7 @@ extern CommunicatorClass Communicator;
 extern std::shared_ptr<UDPNameserver> N;
 extern vector<std::shared_ptr<UDPNameserver> > g_udpReceivers;
 extern int avg_latency;
-extern TCPNameserver *TN;
+extern std::shared_ptr<TCPNameserver> TN;
```
Comment (rgacogne) [inline pdns/common_startup.hh]:
```suggestion
extern std::unique_ptr<TCPNameserver> TN;
```

Comment (pieterlexis) [reply pdns/common_startup.hh]: I made it a `shared_ptr` as the `UDPNameserver` is also a `shared_ptr`, but it might indeed be unique

Comment (rgacogne) [reply pdns/common_startup.hh]: I haven't looked into why but `UDPNameserver` objects are apparently also stored in a vector so they can't be unique, but `TCPNameserver` ones should be fine.

File: pdns/receiver.cc [hunk 2]
```diff
@@ -590,8 +590,7 @@ int main(int argc, char **argv)
       }
     }
 
-    if(!::arg().mustDo("disable-tcp"))
-      TN=new TCPNameserver; 
+    TN = std::make_unique<TCPNameserver>();
```
Comment (rgacogne) [inline pdns/receiver.cc]: C++11 doesn't provide `make_unique` in the `std` namespace, we do have our own `make_unique` in `misc.hh`, though.

**Pull request merged**